arangod/Cluster/ServerState.cpp

```diff
@@ -331,6 +336,27 @@ bool ServerState::unregister() {
   return r.successful();
 }
 
+uint64_t ServerState::readRebootId(AgencyComm& comm)
+{
+  TRI_ASSERT(!_id.empty());
+  std::string rebootIdPath = currentServersRegisteredPref + _id + "/rebootId";
+  AgencyCommResult result = comm.getValues(rebootIdPath);
+
+  if (!result.successful()) {
+    LOG_TOPIC("762ed", WARN, Logger::CLUSTER)
+      << "Could not read back " << rebootIdPath;
+
+    return 0;
+  }
+
+  auto slicePath = AgencyCommManager::slicePath(rebootIdPath);
+  auto valueSlice = result.slice()[0].get(slicePath);
+
+  TRI_ASSERT(valueSlice.isInteger());
```

Comment:
In the above case we return `0` and not handle the error.
In this case we will throw an error, if we do not get a numeric value (e.g. NONE instead). and not return 0.

Is this correct behavior?
Also if I am not mistaken, let us a assume a network hiccup in the above call => we get some error, which will be dropped, but we will return `0` in this case.
Assume further the last value was `0` (or `1` whatever will end up AFTER the first boot of a server)
=> Do we distinguish first boot, vs. second boot properly here?

Reply:
I can answer my question, I think we do distinguish with the below transaction.

However I think the error case here is not handled properly in ProductionMode.
If we cannot read from Agency (after successfully writing to it) we will end up with a bootID 0 which differs from the bootID in agency, and I guess this is no good.
So I guess we need to make sure to return 0 in every error case here.
And handle it in the code below.